engines/macventure/macventure.cpp

```diff
@@ -0,0 +1,1184 @@
+			old.exitx != _world->getObjAttr(id, kAttrExitX) ||
+			old.exity != _world->getObjAttr(id, kAttrExitY))
+			_gui->updateExit(id);
+	}
+	WindowReference win = getObjWindow(id);
+	ObjID cur = id;
+	ObjID root = _world->getObjAttr(1, kAttrParentObject);
+	while (cur != root)	{
+		if (cur == 0 || !_world->getObjAttr(cur, kAttrContainerOpen)) break;
+		cur = _world->getObjAttr(cur, kAttrParentObject);
+	}
+	if (cur == root) {
+		if (win) return;
+		enqueueObject(kOpenWindow, id); //open
+	} else {
+		if (!win) return;
+		enqueueObject(kCloseWindow, id); //close
+	}
+
+	// Update children
+	Common::Array<ObjID> children = _world->getChildren(id, true);
+	for (uint i = 0; i < children.size(); i++) {
+		enqueueObject(kUpdateObject, children[i]);
+	}
+}
+
+void MacVentureEngine::reflectSwap(ObjID fromID, ObjID toID) {
+	//warning("reflectSwap: untested");
+	WindowReference from = getObjWindow(fromID);
+	WindowReference to = getObjWindow(toID);
+	WindowReference tmp = to;
+	debugC(1, kMVDebugMain, "Swap Object[%d] to Object[%d], from win[%d] to win[%d] ",
+		fromID, toID, from, to);
+
+	if (!to) {
+		tmp = from;
+	}
+	if (tmp) {
+		Common::String newTitle = _world->getText(toID, 0, 0); // Ignores src and targ in the original
+		_gui->setWindowTitle(tmp, newTitle);
+		_gui->updateWindowInfo(tmp, toID, _world->getChildren(toID, true));
+		updateWindow(tmp);
+	}
+}
+
+void MacVentureEngine::toggleExits() {
+	Common::Array<ObjID> exits = _currentSelection;
+	while (!exits.empty()) {
+		ObjID obj = exits.front();
+		exits.remove_at(0);
+		highlightExit(obj);
+		updateWindow(findParentWindow(obj));
+	}
+}
+
+void MacVentureEngine::zoomObject(ObjID objID) {
+	warning("zoomObject: unimplemented");
+}
+
+bool MacVentureEngine::isObjEnqueued(ObjID objID) {
+	Common::Array<QueuedObject>::const_iterator it;
+	for (it = _objQueue.begin(); it != _objQueue.end(); it++) {
+		if ((*it).object == objID) return true;
+	}
+	return false;
+}
+
+bool MacVentureEngine::isGameRunning() {
+	return (_gameState == kGameStateInit || _gameState == kGameStatePlaying);
+}
+
+ControlAction MacVenture::MacVentureEngine::referenceToAction(ControlType id) {
+	switch (id) {
+	case MacVenture::kControlExitBox:
+		return kActivateObject;//??
+	case MacVenture::kControlExamine:
+		return kExamine;
+	case MacVenture::kControlOpen:
+		return kOpen;
+	case MacVenture::kControlClose:
+		return kClose;
+	case MacVenture::kControlSpeak:
+		return kSpeak;
+	case MacVenture::kControlOperate:
+		return kOperate;
+	case MacVenture::kControlGo:
+		return kGo;
+	case MacVenture::kControlHit:
+		return kHit;
+	case MacVenture::kControlConsume:
+		return kConsume;
+	default:
+		return kNoCommand;
+	}
+}
+
+// Data retrieval
+
+bool MacVentureEngine::isPaused() {
+	return _paused;
+}
+
+bool MacVentureEngine::needsClickToContinue() {
+	return _clickToContinue;
+}
+
+Common::String MacVentureEngine::getCommandsPausedString() const {
+	return Common::String("Click to continue");
```

Comment:
I think there are no versions of the games in other languages, so unless we were to create the assets ourselves, I don't think it could be played in other languages.